src/main/java/com/example/survey/controllers/SurveyController.java

```diff
@@ -15,32 +20,31 @@ public SurveyController(SurveyRepo surveyRepo) {
 
     @GetMapping("/survey")
     public String survey(@RequestParam("id") long id, Model model) {
-        Optional<Survey> surveyRepo = this.surveyRepo.findById(id);
-        model.addAttribute("survey", surveyRepo.get());
+        Survey survey = this.surveyRepo.findById(id);
+        model.addAttribute("survey", survey);
         return "survey";
     }
 
     @PostMapping("/survey")
     @ResponseBody
     public Survey getSurvey(@RequestParam("id") long id) {
-        Optional<Survey> surveyRepo = this.surveyRepo.findById(id);
-        return surveyRepo.get();
+        Survey survey = this.surveyRepo.findById(id);
+        return survey;
     }
 
-    @PostMapping("/new")
+    @PostMapping(value = "/new", produces = "application/json; charset=utf-8", consumes = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
-    public Survey newSurvey(@RequestParam("name") String name) {
-        Survey s = new Survey(name);
-        this.surveyRepo.save(s);
-        return s;
+    public Survey newSurvey(@RequestBody Survey survey) {
```

Comment:
Does this mean the question will come in the form of the following and be decoding into the concrete type? 
```
"questions": [
   { "name": "joe", "type": "text", ... }
]
```

Reply:
yeah! for the FE, will have to have an add question box on top of the name input, and maybe a radio list for the type of questions